python/lsst/analysis/drp/makeQaTractTables.py

```diff
@@ -0,0 +1,326 @@
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import pandas as pd
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+
+__all__ = ["MakeForcedQaTractTablesTaskConfig", "MakeForcedQaTractTablesTask",
+           "MakeUnforcedQaTractTablesTaskConfig", "MakeUnforcedQaTractTablesTask"]
+
+
+class BaseMakeQaTractTablesTaskConnections(pipeBase.PipelineTaskConnections,
+                                           defaultTemplates={"coaddName": "deep"},
+                                           dimensions=("tract", "skymap")):
+    inputObjCats = cT.Input(
+        doc="The coadd catalog from which to collect relevant columns.",
+        storageClass="DataFrame",
+        name="{coaddName}Coadd_obj",
+        dimensions=("tract", "patch", "skymap"),
+        multiple=True,
+        deferLoad=True,
+    )
+    inputObjectTable = cT.Input(
+        doc=("The objectTable_tract associated with the tract.  Only used to ensure the tables persisted "
+             "have the same row ordering as the objectTable_tract tables for ease of joint use."),
+        storageClass="DataFrame",
+        name="objectTable_tract",
+        dimensions=("tract", "skymap"),
+        deferLoad=True,
+    )
+
+
+class BaseMakeQaTractTablesTaskConfig(pipeBase.PipelineTaskConfig,
+                                      pipelineConnections=BaseMakeQaTractTablesTaskConnections):
+    filterMap = pexConfig.DictField(
+        keytype=str, itemtype=str,
+        default={},
+        doc=("Dictionary mapping canonical band to the camera's physical filter.  This is currently "
+             "required for {coaddName}Coadd_obj that are indexed on the latter as the dataId supplied "
+             "will only contain the former (this will likely become unnecessary once the postprocess "
+             "scripts are updated to index on band).")
+    )
+    dfDataset = pexConfig.ChoiceField(
+        dtype=str,
+        default="forced",
+        allowed={
+            "forced": "Measurement values come from the forced_src catalogs",
+            "unforced": "Measurement values come from the meas catalogs",
+        },
+        doc="Type of measurement values of interest, either forced or unforced."
+    )
```

Comment:
Shouldn't need this option either; it's effectively fixed to a constant by the subclass.

We can at least just make it a class-level constant in the subclass tasks, but I bet we can do better; I'll see if I can make a more concrete suggestion when we get down to that code.